app/views/handlers/list_action.py

```diff
@@ -67,10 +67,13 @@ def get_next_location_url(self):
         if url := self.get_section_or_final_summary_url():
             return url
 
-        if self.router.is_block_complete(
-            block_id=self.parent_location.block_id,
-            section_id=self.parent_location.section_id,
-            list_item_id=self.parent_location.list_item_id,
+        if (
+            self.router.is_block_complete(
+                block_id=self.parent_location.block_id,
+                section_id=self.parent_location.section_id,
+                list_item_id=self.parent_location.list_item_id,
+            )
+            and not self.parent_block["type"] == "ListCollectorContent"
```

Comment:
Why are we checking the type here?.

Reply:
If repeating block is complete, parent list collector content block is also complete and this will try to get you to next location url (if this check is removed first "if" will run as all conditions will be valid) **but** we want to get to the second "return" and go to parent location url which will be the list collector content block. So this was added just to have the correct journey.